src/hotspot/share/runtime/objectMonitor.cpp

```diff
@@ -2036,66 +2037,67 @@ void ObjectMonitor::wait(jlong millis, bool interruptible, TRAPS) {
 
 bool ObjectMonitor::notify_internal(JavaThread* current) {
   bool did_notify = false;
-  Thread::SpinAcquire(&_wait_set_lock);
-  ObjectWaiter* iterator = dequeue_waiter();
-  if (iterator != nullptr) {
-    guarantee(iterator->TState == ObjectWaiter::TS_WAIT, "invariant");
-
-    if (iterator->is_vthread()) {
-      oop vthread = iterator->vthread();
-      java_lang_VirtualThread::set_notified(vthread, true);
-      int old_state = java_lang_VirtualThread::state(vthread);
-      // If state is not WAIT/TIMED_WAIT then target could still be on
-      // unmount transition, or wait could have already timed-out or target
-      // could have been interrupted. In the first case, the target itself
-      // will set the state to BLOCKED at the end of the unmount transition.
-      // In the other cases the target would have been already unblocked so
-      // there is nothing to do.
-      if (old_state == java_lang_VirtualThread::WAIT ||
+  {
```

Comment:
I don't think this extra level of block scope is needed. The only thing outside the end of this
extra scope is the `return did_notify`, which could just as well be inside. Your call...